spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/KubernetesClientConfigDataLocationResolver.java

```diff
@@ -83,20 +82,14 @@ protected void registerBeans(ConfigDataLocationResolverContext resolverContext,
 		}
 	}
 
-	private CoreV1Api registerClientAndCoreV1Api(ConfigurableBootstrapContext bootstrapContext,
+	private CoreV1Api registerCoreV1Api(ConfigurableBootstrapContext bootstrapContext,
 			KubernetesClientProperties kubernetesClientProperties) {
 		ApiClient apiClient = kubernetesApiClient();
 		apiClient.setUserAgent(kubernetesClientProperties.userAgent());
-		registerSingle(bootstrapContext, ApiClient.class, apiClient, "configDataApiClient");
```

Comment:
we only need the `ApiClient` in order to create and register `CoreV1Api`. As such : do not register it and rename method to reflect that.